src/main.zig

```diff
@@ -1267,9 +1267,17 @@ fn buildOutputType(
     }
 
     if (link_objects.items.len == 0 and root_src_file == null and
-        c_source_files.items.len == 0 and arg_mode == .run)
+        c_source_files.items.len == 0 and !show_builtin)
     {
-        fatal("`zig run` expects at least one positional argument", .{});
+        switch (arg_mode) {
+            .run => fatal("`zig run` expects at least one positional argument", .{}),
+            .build => |build_type| switch (build_type) {
+                .Exe => fatal("`zig build-exe` expects at least one positional argument", .{}),
+                .Lib => fatal("`zig build-lib` expects at least one positional argument", .{}),
+                .Obj => fatal("`zig build-obj` expects at least one positional argument", .{}),
+            },
+            else => {},
```

Comment:
why not the similar message for e.g. `translate-c`

Reply:
It already has one:
```
$ zig translate-c
error: translate-c expects exactly 1 source file (found 0)
```
We could do it here to make it more consistent, but I think it is unnecessary.